test/test_timing.py

```diff
@@ -0,0 +1,157 @@
+from types import SimpleNamespace
+
+import pytest
+from mcap_ros1._vendor.genpy.rostime import Duration as ROS1Duration
+from mcap_ros1._vendor.genpy.rostime import Time as ROS1Time
+
+from kappe.module.timing import fix_ros1_time
+
+
+def test_fix_ros1_time_basic():
+    """Test fixing a simple ROS1 Time object."""
+    msg = SimpleNamespace()
+    msg.__slots__ = ['timestamp']
+    msg.timestamp = ROS1Time(secs=10, nsecs=500000000)
+
+    fix_ros1_time(msg)
+
+    assert hasattr(msg.timestamp, 'sec')
+    assert hasattr(msg.timestamp, 'nanosec')
+    assert msg.timestamp.sec == 10
+    assert msg.timestamp.nanosec == 500000000
+
+
+def test_fix_ros1_duration_basic():
+    """Test fixing a simple ROS1 Duration object."""
+    msg = SimpleNamespace()
+    msg.__slots__ = ['duration']
+    msg.duration = ROS1Duration(secs=5, nsecs=250000000)
+
+    fix_ros1_time(msg)
+
+    assert hasattr(msg.duration, 'sec')
+    assert hasattr(msg.duration, 'nanosec')
+    assert msg.duration.sec == 5
+    assert msg.duration.nanosec == 250000000
+
+
+def test_fix_ros1_time_nested():
+    """Test fixing nested messages with ROS1 Time (recursively converts all nested objects)."""
+    inner = SimpleNamespace()
+    inner.__slots__ = ['timestamp']
+    inner.timestamp = ROS1Time(secs=20, nsecs=100000000)
+
+    outer = SimpleNamespace()
+    outer.__slots__ = ['header', 'time']
+    outer.header = inner
+    outer.time = ROS1Time(secs=10, nsecs=50000000)
+
+    fix_ros1_time(outer)
+
+    # Top-level ROS1Time is converted
+    assert hasattr(outer.time, 'sec')
```

Comment:
I guess not needed but having just one seems weird 

```suggestion
    assert hasattr(outer.time, 'sec')
    assert hasattr(outer.time, 'nanosec')
```